### PR TITLE
feat(APIGuildForum): add support for forums, part 1

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Add initial support for forums given documentation in the docs PR.

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**

- https://github.com/discord/discord-api-docs/pull/4761

### CODE REVIEW

File: deno/payloads/v10/channel.ts [hunk 1]
```diff
@@ -223,6 +225,8 @@ export interface APIThreadChannel
 	last_message_id?: Snowflake | null;
 }
 
+export type APIGuildForumChannel = APIGuildTextChannel<ChannelType.GuildForum>;
```
Comment (vladfrangu) [inline deno/payloads/v10/channel.ts:228]: Should we name them APIForumChannel instead? (considering APINewsChannel)... I'd say no, but I want thoughts

File: deno/rest/common.ts [hunk 2]
```diff
@@ -126,6 +130,7 @@ export enum RESTJSONErrorCodes {
 	ApplicationCommandWithThatNameAlreadyExists = 40041,
 
 	InteractionHasAlreadyBeenAcknowledged = 40060,
+	TagNamesMustBeUnique,
```
Comment (vladfrangu) [inline deno/rest/common.ts:133]: Isn't it TagName?

File: deno/rest/v10/mod.ts [hunk 3]
```diff
@@ -834,6 +834,14 @@ export const Routes = {
 	guildScheduledEventUsers(guildId: Snowflake, guildScheduledEventId: Snowflake) {
 		return `/guilds/${guildId}/scheduled-events/${guildScheduledEventId}/users` as const;
 	},
+
+	/**
+	 * Route for
+	 * - POST /channels/${channel.id}/threads
+	 */
+	guildForumThreads(channelId: Snowflake) {
```
Comment (vladfrangu) [inline deno/rest/v10/mod.ts:842]: Idk how to feel about this, is this not a route already? if it is, alias it please

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/payloads/v10/channel.ts (hunk 1 above)
Comment (suneettipirneni) [reply deno/payloads/v10/channel.ts:228]: The type is documented as `GUILD_FORUM` so if we wanna stay in line with API naming conventions it should be `APIGuildForumChannel` imo.

File: deno/rest/common.ts (hunk 2 above)
Comment (suneettipirneni) [reply deno/rest/common.ts:133]: Docs state it like this:
> Tag names must be unique

File: deno/payloads/v10/channel.ts (hunk 1 above)
Comment (vladfrangu) [reply deno/payloads/v10/channel.ts:228]: Bleh, we should probably rename news channels sometime then...

**Comment by vladfrangu** [comment]
Tis needs a rebase

**Comment by suneettipirneni** [comment]
Upstream PR landed

File: deno/payloads/v10/channel.ts [hunk 4]
```diff
@@ -1358,6 +1367,13 @@ export interface APITextInputComponent extends APIBaseComponent<ComponentType.Te
 	required?: boolean;
 }
 
+/**
+ * https://discord.com/developers/docs/resources/channel/channel-flags
```
Comment (vitorlops) [inline deno/payloads/v10/channel.ts:1371]:
```suggestion
 * https://discord.com/developers/docs/resources/channel#channel-object-channel-flags
```

File: deno/payloads/v9/channel.ts [hunk 5]
```diff
@@ -1358,6 +1367,13 @@ export interface APITextInputComponent extends APIBaseComponent<ComponentType.Te
 	required?: boolean;
 }
 
+/**
+ * https://discord.com/developers/docs/resources/channel/channel-flags
```
Comment (vitorlops) [inline deno/payloads/v9/channel.ts:1371]:
```suggestion
 * https://discord.com/developers/docs/resources/channel#channel-object-channel-flags
```

File: payloads/v10/channel.ts [hunk 6]
```diff
@@ -1358,6 +1367,13 @@ export interface APITextInputComponent extends APIBaseComponent<ComponentType.Te
 	required?: boolean;
 }
 
+/**
+ * https://discord.com/developers/docs/resources/channel/channel-flags
```
Comment (vitorlops) [inline payloads/v10/channel.ts:1371]:
```suggestion
 * https://discord.com/developers/docs/resources/channel#channel-object-channel-flags
```

File: payloads/v9/channel.ts [hunk 7]
```diff
@@ -1358,6 +1367,13 @@ export interface APITextInputComponent extends APIBaseComponent<ComponentType.Te
 	required?: boolean;
 }
 
+/**
+ * https://discord.com/developers/docs/resources/channel/channel-flags
```
Comment (vitorlops) [inline payloads/v9/channel.ts:1371]:
```suggestion
 * https://discord.com/developers/docs/resources/channel#channel-object-channel-flags
```

**Review by vitorlops — CHANGES_REQUESTED** [review]
Wrong links.

File: deno/rest/v9/channel.ts [hunk 8]
```diff
@@ -595,6 +595,12 @@ export type RESTPostAPIChannelMessagesThreadsJSONBody = AddUndefinedToPossiblyUn
 	rate_limit_per_user?: number;
 }>;
 
+/**
+ * https://discord.com/developers/docs/resources/channel#start-thread-in-forum-channel
+ */
+export type RESTPostAPIGuildForumThreadsJSONBody = RESTPostAPIChannelMessagesThreadsJSONBody &
```
Comment (vladfrangu) [inline deno/rest/v9/channel.ts:601]: document form data too

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/payloads/v9/channel.ts [hunk 9]
```diff
@@ -234,7 +238,8 @@ export type APIChannel =
 	| APIVoiceChannel
 	| APIGuildCategoryChannel
 	| APIThreadChannel
-	| APINewsChannel;
+	| APINewsChannel
+	| APIGuildCategoryChannel;
```
Comment (vitorlops) [inline deno/payloads/v9/channel.ts:242]:
```suggestion
	| APIGuildForumChannel;
```

File: payloads/v9/channel.ts [hunk 10]
```diff
@@ -234,7 +238,8 @@ export type APIChannel =
 	| APIVoiceChannel
 	| APIGuildCategoryChannel
 	| APIThreadChannel
-	| APINewsChannel;
+	| APINewsChannel
+	| APIGuildCategoryChannel;
```
Comment (vitorlops) [inline payloads/v9/channel.ts:242]:
```suggestion
	| APIGuildForumChannel;
```

**Review by vitorlops — CHANGES_REQUESTED** [review]
Duplicated types. 👀

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**